### PR TITLE
add correct return types in JSDoc

### PR DESCRIPTION
**Pull request by 43081j**
For PolymerLabs/gen-typescript-declarations#69.

Introduces some jsdoc so we produce the correct types.

### CODE REVIEW

**Comment by notwaldorf** [comment]
@rictic does this look ok?

File: iron-form.html [hunk 1]
```diff
@@ -316,6 +325,9 @@
 
       /**
        * Resets the form to the default values.
+       *
+       * @param {!Event} event
```
Comment (rictic) [inline iron-form.html]: Looks like this should be `Event=` instead, as it reads like an optional parameter.

File: iron-form.html [hunk 2]
```diff
@@ -272,6 +278,9 @@
 
       /**
        * Submits the form.
+       *
+       * @param {!Event} event
```
Comment (rictic) [inline iron-form.html]: Looks like this should be `Event=` instead, as it reads like an optional parameter.

**Comment by 43081j** [comment]
you were right @rictic i was blind to the truthy check, sorted it now :)

**Pull request merged**